File: Selenium/Activities/Activity14.java
package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;

import java.util.List;

public class Activity14 {
    public static void main(String[] args) {
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "/dev/null");
        WebDriverManager.firefoxdriver().setup();
        WebDriver driver = new FirefoxDriver();
        driver.get("https://v1.training-support.net/selenium/tables");
        System.out.println("Title of webpage:" + driver.getTitle());

        List<WebElement> rows = driver.findElements(By.xpath("//table[@id='sortableTable']/tbody/tr"));
        System.out.println("Number of rows in first table:"+rows.size());

        List<WebElement> columns=driver.findElements(By.xpath("//table[@id='sortableTable']/tbody/tr[1]/td"));
        System.out.println("Number of columns in first table:"+columns.size());
        System.out.println("Cell value in second row and column:"+driver.findElement(By.xpath("//table[@id='sortableTable']/tbody/tr[2]/td[2]")).getText());
        System.out.println("Clicking on header of the first column...");
        driver.findElement(By.xpath("//table[@id='sortableTable']/thead/tr/th[1]")).click();
        System.out.println("Cell value in second row and column after sorting by name:"+driver.findElement(By.xpath("//table[@id='sortableTable']/tbody/tr[2]/td[2]")).getText());

        List<WebElement> footer=driver.findElements(By.xpath("//table[@id='sortableTable']/tfoot/tr/th"));
        System.out.println("Cell values of the footer: ");
        for(WebElement cell:footer) {
            System.out.println(cell.getText());
        }
        driver.close();
    }
}

File: Selenium/Activity14.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from selenium.webdriver.support.wait import WebDriverWait
from webdriver_manager.firefox import GeckoDriverManager

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    driver.get("https://v1.training-support.net/selenium/tables")
    wait = WebDriverWait(driver, 10)
    print("Title of webpage:" + driver.title)
    rows = driver.find_elements(By.XPATH, "//table[@id='sortableTable']/tbody/tr")
    print("Number of rows in first table:" + str(len(rows)))
    columns = driver.find_elements(By.XPATH, "//table[@id='sortableTable']/tbody/tr[1]/td")
    print("Number of columns in first table:" + str(len(columns)))
    print("Cell value in second row and column:" + driver.find_element(By.XPATH,"//table[@id='sortableTable']/tbody/tr[2]/td[2]").text)
    print("Clicking on header of the first column...")
    driver.find_element(By.XPATH, "//table[@id='sortableTable']/thead/tr/th[1]").click()
    print("Cell value in second row and column after sorting by name:" + driver.find_element(By.XPATH,"//table[@id='sortableTable']/tbody/tr[2]/td[2]").text)
    footer = driver.find_elements(By.XPATH, "//table[@id='sortableTable']/tfoot/tr/th")
    print("Cell values of the footer: ")
    for cell in footer:
        print(cell.text)
    driver.close()
